server/src/com/cloud/user/AccountManagerImpl.java

```diff
@@ -2472,6 +2464,11 @@ public void buildACLSearchParameters(Account caller, Long id, String accountName
                 if (projectId.longValue() == -1) {
                     if (caller.getType() == Account.ACCOUNT_TYPE_NORMAL) {
                         permittedAccounts.addAll(_projectMgr.listPermittedProjectAccounts(caller.getId()));
+
+                        //permittedAccounts can be empty when the caller is not a part of any project (a domain account)
+                        if (permittedAccounts.isEmpty()) {
```

Comment:
Is it possible for this `permittedAccounts`  be null?
If not, the code is ok, otherwise it might be a good idea to use `CollectionUtils.isEmpty`

Reply:
There are multiple places which call `AccountManagerImpl#buildACLSearchParameters` and all of them have `List<Long> permittedAccounts = new ArrayList<Long>();`. So technically there's no need to check for null or use `CollectionUtils`.

Reply:
and yes, I'm using the same code formatter (checked in the repo) in Eclipse and not the `eclipse.epf`.

Reply:
thanks!
So, it seems that the code was not formatted properly before.

Reply:
I would say so.